SpatialGDK/Source/SpatialGDKTests/SpatialGDK/EngineClasses/SpatialVirtualWorkerTranslator/SpatialVirtualWorkerTranslatorTest.cpp

```diff
@@ -1,41 +1,81 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
+#include "LBStrategySpy.h"
+
 #include "Tests/TestDefinitions.h"
 
 #include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "SpatialConstants.h"
 #include "Utils/SchemaUtils.h"
-#include "UObject/UObjectGlobals.h"
 
-#include "CoreMinimal.h"
+#include "Templates/UniquePtr.h"
+#include "UObject/UObjectGlobals.h"
 
+#include <WorkerSDK/improbable/c_worker.h>
 #include <WorkerSDK/improbable/c_schema.h>
 
 #define VIRTUALWORKERTRANSLATOR_TEST(TestName) \
 	GDK_TEST(Core, SpatialVirtualWorkerTranslator, TestName)
 
-VIRTUALWORKERTRANSLATOR_TEST(Given_init_is_not_called_THEN_return_not_ready)
+VIRTUALWORKERTRANSLATOR_TEST(GIVEN_init_is_not_called_THEN_return_not_ready)
+{
+	ULBStrategySpy* LBStrategySpy = NewObject<ULBStrategySpy>();
+	TUniquePtr<SpatialVirtualWorkerTranslator> Translator = MakeUnique<SpatialVirtualWorkerTranslator>(LBStrategySpy, SpatialConstants::TRANSLATOR_UNSET_PHYSICAL_NAME);
+
+	TestFalse("Translator without local virtual worker ID is not ready.", Translator->IsReady());
+	TestTrue("LBStrategy spy reports an invalid virtual worker ID.", LBStrategySpy->GetVirtualWorkerId() == SpatialConstants::INVALID_VIRTUAL_WORKER_ID);
```

Comment:
If you really want to make this a spy, you can record whether SetLocalVirtualWorkerId has been called on the LBStrategy.

Reply:
Actually, nevermind I had forgotten that the Abstract has an implementation for SetLocalVirtualWorkerId, so that's what you're doing here. Which is also just odd for an Abstract class to have that, and not to have the Get implementation provided also.

Reply:
Have changed to stub, I think checking whether the value has been changed to our expected value from the unintialized one is sufficient?